.golangci.yml

```diff
@@ -91,7 +91,7 @@ linters-settings:
     - 'os\.Stderr(# Disallowed output streams used\. Use environment\.InputOutput\(\).Out instead\.)?'
     - 'os\.Stdin(# Disallowed output streams used\. Use environment\.InputOutput\(\).Out instead\.)?'
   staticcheck:
-    go: "1.16"
+    go: "1.18"
```

Comment:
Maybe it makes sense to factor out this as well as the changes in `qa-tests-backend/` and `tools/` into a separate PR. I assume we want to do some more thorough investigation as to how the set changes impact the performance, but the aforementioned changes block usage of go1.18 features otherwise (at least for some use cases). I probably should've done this when bumping the version in `go.mod`, but unfortunately I missed that.

Reply:
@misberner, would you suggest me to run some performance tests on the generic set versus generated?
Do you expect a noticeable degradation, or just to be sure there's none?

Reply:
Well, I would suggest to start with a microbenchmark (`_bench_test.go`) to assess if there is much of a performance impact at all. If there's none, or none that is significant, then no, I don't think we need to do any further tests. However, if a microbenchmark suggests that there is some noticeable degradation, we have to see further.